kindelia/src/cli.rs

```diff
@@ -120,6 +121,9 @@ pub enum CliCommand {
     /// Whether to consider size and mana in the execution.
     #[clap(long)]
     sudo: bool,
+    /// Network id / magic number.
+    #[clap(long, value_parser=maybe_hex::<u32>)]
+    network_id: Option<u32>,
```

Comment:
Maybe it makes more sense to accept the "prelude" code itself here instead of the network id.
"What does offline testing has to do with the network id?"

Reply:
> Maybe it makes more sense to accept the "prelude" code itself here instead of the network id.

So accept the code itself as a CLI argument?   And in the config file?  That's a potentially very long and multiline string...   I don't see it.

> What does offline testing has to do with the network id?

I wondered this myself.   But this code eventually calls `hvm::test_statements()` which has a call `parse_code(GENESIS_CODE)`.   And `GENESIS_CODE` is defined as `include_str!("../genesis.kdl")`. 

In other words, this `check` command is relying on GENESIS_CODE exactly as Node::new() does.  And GENESIS_CODE is the thing we are trying to make dynamic per network, as I understand it.

I did not look into it deeply, but rather assumed that the tests may rely on the prelude/genesis statements.

So I guess I would ask you the same question.  Why is this needed?

Reply:
ok, I see why it is using the genesis code.  The prelude is setting up required basics.  

I found that `test_statements()` is passing the parsed code to `init_runtime()`.   I tried passing an empty vec instead.  Then I ran `kindelia test example/example.kdl`:

```
$ kindelia test ../example/example.kdl 
Block 
=====

0000000000 [run] ERROR: 'Call' is not defined.
```

So it seems to be truly necessary.  I will get rid of `network_id` and pass `&[Statement]`.

Reply:
not marking this resolved yet, because as I see it, `--network-id` needs to be an argument for `kindelia check` because the test statements typically depend on the prelude, which varies by network.

@steinerkelvin do you agree?

Reply:
> So accept the code itself as a CLI argument?

A file containing the code. If not provided it would use the `include_str!("../genesis.kdl")` on the binary (see my comment).

> --network-id needs to be an argument for kindelia check

I agree! `kindelia check` will be a typically online command (it would check if names are taken etc), it needs to talks with a node and so is aware of the network id.
But `kindelia test` should be offline, so if it takes at most a parameter that would be something like a `--prelude` one, that would point to a file with alternative initial statements. The default would be to use the `genesis.kdl` included in the binary — we should avoid reading files as possible.

Reply:
I modified `kindelia test` to accept a `--genesis` arg that allows user to provide a path to  prelude file, or else it defaults to `kindelia_core/genesis-tests.kdl` which is compiled into the executable.